pkg/apis/core/helper/helpers.go

```diff
@@ -167,3 +169,61 @@ func FindWorkerByName(workers []core.Worker, name string) *core.Worker {
 	}
 	return nil
 }
+
+// GetRemovedVersions finds versions that have been removed in the old compared to the new version slice.
+// returns a map associating the index in the old version slice with the version and a string set containing the versions.
+func GetRemovedVersions(old []core.ExpirableVersion, new []core.ExpirableVersion) (map[int]core.ExpirableVersion, sets.String) {
+	return getVersionDiff(old, new)
+}
+
+// GetAddedVersions finds versions that have been added in the new compared to the new version slice.
+// returns a map associating the index in the new version slice with the version and a string set containing the versions.
+func GetAddedVersions(old []core.ExpirableVersion, new []core.ExpirableVersion) (map[int]core.ExpirableVersion, sets.String) {
+	return getVersionDiff(new, old)
+}
+
+// getVersionDiff gets versions that are in v1 but not in v2.
+// Returns versions mapped to its index in v1 and also a string set containing the versions
+func getVersionDiff(v1, v2 []core.ExpirableVersion) (map[int]core.ExpirableVersion, sets.String) {
```

Comment:
Why don't you only return a `map[string]int`, so you can simply check, if a given version was added/removed, and get the index for the field path?
```suggestion
func getVersionDiff(v1, v2 []core.ExpirableVersion) (map[string]int) {
```